### PR TITLE
*: add a new I_S table and support show create for resource groups

### LINKED ISSUE #39781: Add a memory table to show Resource Groups @tiancaiamao

### PR DESCRIPTION
**Pull request by BornChanger**
### What problem does this PR solve?

Issue Number: Close #39781

Problem Summary:

### What is changed and how it works?

1.  A new mem table information_schema.resource_groups
2. Support to `show create resource group` command
3. Fix problem of binding user to non existent resource group

### Check List

Tests 

- [X] Unit test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/infoschema_reader.go [hunk 1]
```diff
@@ -3386,6 +3388,23 @@ func (e *memtableRetriever) setDataFromPlacementPolicies(sctx sessionctx.Context
...
 }
 
+func (e *memtableRetriever) setDataFromResourceGroups(sctx sessionctx.Context) error {
+	is := sessiontxn.GetTxnManager(sctx).GetTxnInfoSchema()
+	resourceGroups := is.AllResourceGroups()
+	rows := make([][]types.Datum, 0, len(resourceGroups))
+	for _, group := range resourceGroups {
+		row := types.MakeDatums(
+			group.ID,
+			group.Name.L,
```
Comment (glorv) [inline executor/infoschema_reader.go:3398]:
```suggestion
			group.Name.O,
```

Comment (BornChanger) [reply executor/infoschema_reader.go]: fixed

File: executor/show.go [hunk 2]
```diff
@@ -1485,6 +1493,17 @@ func (e *ShowExec) fetchShowCreatePlacementPolicy() error {
 	return nil
 }
 
+// fetchShowCreateResourceGroup composes show create resource group result.
+func (e *ShowExec) fetchShowCreateResourceGroup() error {
+	group, found := e.is.ResourceGroupByName(e.ResourceGroupName)
+	if !found {
+		return infoschema.ErrResourceGroupNotExists.GenWithStackByArgs(e.ResourceGroupName.O)
+	}
+	showCreate := ConstructResultOfShowCreateResourceGroup(group)
```
Comment (tiancaiamao) [inline executor/show.go:1502]: The function is not used by other places, so there is no need to expose it, `constructResultOfShowCreateResourceGroup` would be better.

File: infoschema/tables.go [hunk 3]
```diff
@@ -1586,6 +1589,14 @@ var tableMemoryUsageOpsHistoryCols = []columnInfo{
 	{name: "SQL_TEXT", tp: mysql.TypeVarchar, size: 256},
 }
 
+var tableResourceGroupsCols = []columnInfo{
+	{name: "GROUP_ID", tp: mysql.TypeLonglong, size: 64, flag: mysql.NotNullFlag},
```
Comment (tiancaiamao) [inline infoschema/tables.go:1593]: How is the group ID generated?

**Review by tiancaiamao — APPROVED** [review]

File: infoschema/tables.go (hunk 3 above)
Comment (BornChanger) [reply infoschema/tables.go:1593]: Original DDL job's ID. @nolouch

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**